dex/networks/eth/contracts/genruntime.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+# This script generates the runtime bytecoode for a solidity contract. It is used
+# to compare against the runtime bytecode on chain in order to verify that the
+# expected contract is deployed.
+
+if [ "$#" -ne 1 ]
+then
+  echo "Usage: $0 version" >&2
+  exit 1
+fi
+
+ETH_SWAP_VERSION=$1
+SOLIDITY_FILE=ETHSwapV${ETH_SWAP_VERSION}.sol
+if [ ! -f ./${SOLIDITY_FILE} ]
+then
+    echo "${SOLIDITY_FILE} does not exist" >&2
+    exit 1
+fi
+
+solc --bin-runtime --optimize ${SOLIDITY_FILE} -o .
```

Comment:
It's not clear to me what switches to use here.  These are the output options I see:

```
  --asm                EVM assembly of the contracts.
  --asm-json           EVM assembly of the contracts in JSON format.
  --opcodes            Opcodes of the contracts.
  --bin                Binary of the contracts in hex.
  --bin-runtime        Binary of the runtime part of the contracts in hex.
  --abi                ABI specification of the contracts.
```

The output of `--bin` is larger than `--bin-runtime`.  Which one is better to compare?

Also I'm wondering if the use of optimize (and specific compiler version) are going to make this extremely fragile.

Asking for no changes here... just a conversation.

Reply:
`--bin` returns the creation bytecode. This is what has to be sent in a transaction when you initialize a new smart contract. `--bin-runtime` returns the runtime bytecode, which is the code that ends up being stored on chain. This includes everything in the creation bytecode, except the constructor function and constructor arguments, as they will not need to be used again after initialization.

Why do you say that using a specific compiler version would make this fragile? I would think that not having a specific version would make it fragile. Optimize just applies some simplification rules to the unoptimized bytecode. This is a deterministic operation. If the same compiler version is used to optimize both what is deployed and what is used to compare, then we will get the same results.

Reply:
> Why do you say that using a specific compiler version would make this fragile?

Ah I was thinking about optimize and how there are different levels, but you are right, the same considerations apply there as with compiler version.  I was just referring to how identical contracts compiled with newer solc versions probably would get rejected, but I guess that's the point.